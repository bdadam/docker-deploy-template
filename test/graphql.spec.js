const request = require('supertest');
const app = require('../src/app');

jest.mock('../src/db.js');

describe('GET /', () => {
    it('Works', async () => {
        const query = `query { test(id: "1234") { name, id } }`;
        const { status, body } = await request(app).get(`/api/gql/graphql?query=${encodeURIComponent(query)}`);

        expect(status).toEqual(200);
        expect(body).toMatchSnapshot();
    });

    it('Works with variables', async () => {
        const query = `query Test($id: String) { test(id: $id) { name, id } }`;
        const variables = { id: '2345' };

        const { status, body } = await request(app).get(`/api/gql/graphql?query=${encodeURIComponent(query)}&variables=${encodeURIComponent(JSON.stringify(variables))}`);

        expect(status).toEqual(200);
        expect(body).toMatchSnapshot();
    });
});