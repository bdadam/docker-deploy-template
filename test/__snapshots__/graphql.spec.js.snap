// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GET / Works 1`] = `
Object {
  "data": Object {
    "test": Object {
      "id": "1234",
      "name": "Test name",
    },
  },
}
`;

exports[`GET / Works with variables 1`] = `
Object {
  "data": Object {
    "test": Object {
      "id": "2345",
      "name": "Test name",
    },
  },
}
`;
